tls/s2n_early_data_io.c

```diff
@@ -180,7 +180,8 @@ S2N_RESULT s2n_send_early_data_impl(struct s2n_connection *conn, const uint8_t *
      * We only care about the result of this call if it fails. */
     uint32_t early_data_to_send = 0;
     RESULT_GUARD_POSIX(s2n_connection_get_remaining_early_data_size(conn, &early_data_to_send));
-    early_data_to_send = MIN(data_len, early_data_to_send);
+    RESULT_ENSURE_GTE(data_len, 0);
+    early_data_to_send = MIN((uint64_t) data_len, early_data_to_send);
```

Comment:
I regret copying s2n_send and making data_len a ssize_t :/

It might make more sense if you validate data_len at the very beginning of this method, and then assign it to a size_t value for use in the rest of the method.

Same for s2n_recv.